### PR TITLE
UX: New animation for adding and removing reactions

### PR DESCRIPTION
**Pull request by Ahmedgagan**

### CODE REVIEW

**Comment by Ahmedgagan** [comment]
@jjaffeux

File: assets/javascripts/discourse/widgets/discourse-reactions-reaction-button.js.es6 [hunk 1]
```diff
@@ -106,21 +106,26 @@ export default createWidget("discourse-reactions-reaction-button", {
...
+      );
     }
 
     if (currentUserReaction) {
-      return h(`img.btn-toggle-reaction-emoji.btn-icon.no-text`, {
-        src: emojiUrlFor(currentUserReaction.id),
-        alt: `:${currentUserReaction.id}:`
-      });
+      return h(
+        `button.btn-icon.no-text.reaction-button`,
```
Comment (jjaffeux) [inline assets/javascripts/discourse/widgets/discourse-reactions-reaction-button.js.es6]:
```suggestion
        "button.btn-icon.no-text.reaction-button",
```

File: assets/javascripts/discourse/widgets/discourse-reactions-reaction-button.js.es6 [hunk 2]
```diff
@@ -106,21 +106,26 @@ export default createWidget("discourse-reactions-reaction-button", {
     const currentUserReaction = attrs.post.current_user_reaction;
 
     if (hasUsedMainReaction) {
-      return h(`button.btn-toggle-reaction-like.btn-icon.no-text`, [
-        iconNode(mainReactionIcon)
-      ]);
+      return h(
+        `button.btn-toggle-reaction-like.btn-icon.no-text.reaction-button`,
```
Comment (jjaffeux) [inline assets/javascripts/discourse/widgets/discourse-reactions-reaction-button.js.es6]:
```suggestion
        "button.btn-toggle-reaction-like.btn-icon.no-text.reaction-button",
```

File: assets/javascripts/discourse/widgets/discourse-reactions-reaction-button.js.es6 [hunk 3]
```diff
@@ -106,21 +106,26 @@ export default createWidget("discourse-reactions-reaction-button", {
...
+          alt: `:${currentUserReaction.id}:`
+        })
+      );
     }
 
-    return h(`button.btn-toggle-reaction-like.btn-icon.no-text`, [
-      iconNode(`far-${mainReactionIcon}`)
-    ]);
+    return h(
+      `button.btn-toggle-reaction-like.btn-icon.no-text.reaction-button`,
```
Comment (jjaffeux) [inline assets/javascripts/discourse/widgets/discourse-reactions-reaction-button.js.es6]:
```suggestion
      "button.btn-toggle-reaction-like.btn-icon.no-text.reaction-button",
```

**Review by jjaffeux — APPROVED** [review]

**Pull request merged**